x-pack/plugins/observability_ai_assistant/public/components/chat/chat_item.tsx

```diff
@@ -66,28 +87,79 @@ export function ChatItem({
   onStopGeneratingClick,
   onFeedbackClick,
 }: ChatItemProps) {
-  const [isActionsPopoverOpen, setIsActionsPopover] = useState(false);
+  const accordionId = useGeneratedHtmlId({ prefix: 'chat' });
 
-  const handleClickActions = () => {
-    setIsActionsPopover(!isActionsPopoverOpen);
+  const [isPopoverOpen, setIsPopoverOpen] = useState<string | undefined>();
+  const [isAccordionOpen, setIsAccordionOpen] = useState<boolean>(false);
+
+  const handleAccordionToggle = () => {
+    setIsAccordionOpen(!isAccordionOpen);
   };
 
-  const [_, setEditing] = useState(false);
-
-  const actions: ChatItemAction[] = canEdit
-    ? [
-        {
-          id: 'edit',
-          label: i18n.translate('xpack.observabilityAiAssistant.chatTimeline.actions.editMessage', {
-            defaultMessage: 'Edit message',
-          }),
-          handler: () => {
-            setEditing(false);
-            setIsActionsPopover(false);
+  useEffect(() => {
+    const timeout = setTimeout(() => {
+      if (isPopoverOpen) {
+        setIsPopoverOpen(undefined);
+      }
+    }, 800);
+
+    return () => {
+      clearTimeout(timeout);
+    };
+  }, [isPopoverOpen]);
```

Comment:
what's this about?

Reply:
https://github.com/dgieselaar/kibana/assets/535564/bc248f84-10c9-47fd-b49a-121f85ab0b21